src/OpenTK.Core/Platform/Interfaces/IClipboardComponent.cs

```diff
@@ -4,13 +4,65 @@
 using System.Text;
 using System.Threading.Tasks;
 
+#nullable enable
+
 namespace OpenTK.Core.Platform
 {
     // GOAL: We want to support unicode text, rich text and images.
     public interface IClipboardComponent : IPalComponent
     {
+        /// <summary>
+        /// A list of formats that this clipboard component supports.
+        /// </summary>
+        IReadOnlyList<ClipboardFormat> SupportedFormats { get; }
+
+        /// <summary>
+        /// Gets the format of the data currently in the clipboard.
+        /// </summary>
+        /// <returns>The format of the data currently in the clipboard.</returns>
+        ClipboardFormat GetClipboardFormat();
+
+        /// <summary>
+        /// Sets the string currently in the clipboard.
+        /// </summary>
+        /// <param name="text">The text to put on the clipboard.</param>
         void SetClipboardText(string text);
 
-        string GetClipboardText();
+        /// <summary>
+        /// Returns the string currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Text"/> format.
+        /// </summary>
+        /// <returns>The string currently in the clipboard, or null.</returns>
+        string? GetClipboardText();
+
+        /// <summary>
+        /// Gets the audio data currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Audio"/> format.
+        /// </summary>
+        /// <returns>The audio data currently in the clipboard.</returns>
+        // FIXME: Figure out the return type!
+        object? GetClipboardAudio();
+
+        /// <summary>
+        /// Gets the bitmap currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Bitmap"/> format.
+        /// </summary>
+        /// <returns>The bitmap currently in the clipboard.</returns>
+        // FIXME: Figure out the return type!
+        object? GetClipboardBitmap();
+
+        /// <summary>
+        /// Gets the HTML string currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.HTML"/> format.
+        /// </summary>
+        /// <returns>The HTML string currently in the clipboard.</returns>
+        string? GetClipboardHTML();
+
+        /// <summary>
+        /// Gets a list of files currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Files"/> format.
+        /// </summary>
+        /// <returns>The list of files currently in the clipboard.</returns>
+        List<string>? GetClipboardFiles();
```

Comment:
Potentially a list of `System.IO.FileSystemInfo` is more useful than a list of paths, as it could be a directory, link, file or device name.

Reply:
I think it's fine to leave it as strings, trying to create `FileSystemInfo` objects is just another failure point. The input data should be sanitized before trying to parse it, and that should be left to the end user of the API as they are the ones to decide what their threat model is.